### ISSUE TITLE
group order changes when columns are renamed

### ISSUE THREAD

**Issue by sfirke** [tidyverse/dplyr#5078]
I found unexpected behavior in the current version of dplyr and isolated it to one line.  I had janitor tests failing under the new dplyr that previously passed with the current CRAN version.  I resolved it by removing a now-unneeded line of code that caused the behavior -- but I thought I'd report it in case the behavior is of interest or has other implications you see.  If this is nothing, go ahead and close it.

See the output of the two print statements: first the groups are `carb, cyl` then they flip for the second printed data.frame.

```
myfun <- function(dat, ...) {
  expr <- rlang::expr(c(...))
  pos <- tidyselect::eval_select(expr, data = dat)
  print(dat) # look at the group names now...
  names(dat)[pos] <- names(pos) #  I don't need this line, but it causes the groups to reorder
    # I think by column position?
  print(dat) # and again after the renaming
}

> mtcars %>% group_by(carb, cyl) %>% myfun
# A tibble: 32 x 11
# Groups:   carb, cyl [9]
     mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
   <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
 1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
 2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
 3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
 4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
 5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
 6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
 7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
 8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
 9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
# … with 22 more rows
# A tibble: 32 x 11
# Groups:   cyl, carb [9]
     mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
   <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
 1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
 2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
 3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
 4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
 5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
 6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
 7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
 8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
 9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
# … with 22 more rows
> 
```

**Comment by hadley** [comment]
I can't reproduce this with dev dplyr:

``` r
library(dplyr, warn.conflicts = FALSE)

myfun <- function(dat, ...) {
  expr <- rlang::expr(c(...))
  pos <- tidyselect::eval_select(expr, data = dat)
  print(group_vars(dat)) # look at the group names now...
  names(dat)[pos] <- names(pos)
  
  print(group_vars(dat))
}

mtcars %>% group_by(carb, cyl) %>% myfun
#> [1] "carb" "cyl" 
#> [1] "carb" "cyl"
```

<sup>Created on 2020-03-31 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Are you sure you have the very latest version? I suspect it was fixed in #5045

**Comment by sfirke** [comment]
Indeed I was a day before that:
`dplyr       * 0.8.99.9002  2020-03-24 [1] Github (tidyverse/dplyr@35d3ace) `
I should have pulled the latest, sorry!
